myth/Themers/LocalizedViewThemer.php

```diff
@@ -0,0 +1,1165 @@
+    //--------------------------------------------------------------------
+
+    /**
+     * Returns the path to the active/default theme's folder.
+     *
+     * @return string|null
+     */
+    public function getThemePath()
+    {
+        $theme = empty($this->theme) ? $this->default_theme : $this->theme;
+
+        if (! isset($this->folders[ $theme ]))
+        {
+            return null;
+        }
+
+        return $this->folders[$theme];
+    }
+
+    //--------------------------------------------------------------------
+
+
+
+    //--------------------------------------------------------------------
+    // Variants
+    //--------------------------------------------------------------------
+
+    /**
+     * Sets the variant used when creating view names. These variants can
+     * be anything, but by default are used to render specific templates
+     * for desktop, tablet, and phone. The name of the variant is added
+     * to the view name, joined by a "+" symbol.
+     *
+     * Example:
+     *      $this->setVariant('phone');
+     *      $this->display('header');
+     *
+     *      Tries to display "views/header+phone.php"
+     *
+     * @param $variant
+     * @return $this
+     */
+    public function setVariant($variant)
+    {
+        if (isset($this->variants[$variant])) {
+            $this->current_variant = $variant;
+        }
+
+        return $this;
+    }
+    //--------------------------------------------------------------------
+
+    /**
+     * Adds a new variant to the system.
+     *
+     * @param $name
+     * @param $postfix
+     * @return $this|mixed
+     */
+    public function addVariant($name, $postfix)
+    {
+        $this->variants[$name] = $postfix;
+
+        return $this;
+    }
+
+    //--------------------------------------------------------------------
+
+    /**
+     * Removes a variant from the system.
+     *
+     * @param $name
+     * @return $this|mixed
+     */
+    public function removeVariant($name)
+    {
+        if (isset($this->variants[$name])) {
+            unset($this->variants[$name]);
+        }
+
+        return $this;
+    }
+
+    //--------------------------------------------------------------------
+    // Private Methods
+    //--------------------------------------------------------------------
+
+    /**
+     * Handles the actual loading of a view file, and checks for any
+     * overrides in themes, etc.
+     *
+     * @param $view
+     * @param $data
+     *
+     * @return string
+     */
+    private function loadView($view, $data)
+    {
+        // First - does it exist in the current theme?
+        $theme = ! empty($this->active_theme) ? $this->active_theme : $this->default_theme;
+        $theme = ! empty($this->folders[$theme]) ? $this->folders[$theme] : $theme;
+        $theme = rtrim($theme, '/ ') .'/';
+		
+		// Set localization idioms
+		$theme_lang_idom = str_replace(FCPATH .'themes/', FCPATH .'themes/'. CURRENT_LANGUAGE .'/', $theme);
```

Comment:
You cannot rely on FCPATH being an accurate part of the path. The user could specify the theme folder to be anywhere on the system. Checking for 'themes/' should be reliable, but even then, not necessarily. The simplest thing to do is to just tack on the idiom to the theme. We want the 'turkish' language folders to be inside of the theme, anyway.